components/pcloud/pcloud.app.mjs

```diff
@@ -1,11 +1,361 @@
+import get from "lodash/get.js";
+import retry from "async-retry";
+import pcloudSdk from "pcloud-sdk-js";
+
 export default {
   type: "app",
   app: "pcloud",
-  propDefinitions: {},
+  propDefinitions: {
+    fileId: {
+      type: "integer",
+      label: "File ID",
+      async options() {
+        return this.getFileOptions();
+      },
+    },
+    folderId: {
+      type: "integer",
+      label: "Folder ID",
+      async options() {
+        return this.getFolderOptions();
+      },
+      default: 0,
+    },
+    name: {
+      type: "string",
+      label: "Name",
+      description: "Name of the folder to be created.",
+    },
+    overwrite: {
+      type: "boolean",
+      label: "Overwrite?",
+      description: `If true, and an entry with the same name already exists, it will be overwritten.
+            \\
+            Otherwise, an error \`2004\` will be returned instead.`,
+      default: false,
+      optional: true,
+    },
+    showDeleted: {
+      type: "boolean",
+      label: "Show Deleted?",
+      description:
+        "If true, deleted files and folders that can be undeleted will be displayed.",
+      default: false,
+      optional: true,
+    },
+    modifiedTime: {
+      type: "integer",
+      label: "Modified Time",
+      description: "Must be Unix time (seconds).",
+      optional: true,
+    },
+    createdTime: {
+      type: "integer",
+      label: "Created Time",
+      description: `Must be Unix time (seconds).
+      \\
+      Requires \`Modified Time\` to be set.`,
+      optional: true,
+    },
+  },
   methods: {
-    // this.$auth contains connected account data
-    authKeys() {
-      console.log(Object.keys(this.$auth));
+    async api() {
+      global.locationid = Number(this.$auth.locationid);
+      // eslint-disable-next-line no-unused-vars
+      /*
+      const locations = {
+        1: "api.pcloud.com",
+        2: "eapi.pcloud.com",
+      };
+      */
+      return await pcloudSdk.createClient(
```

Comment:
```suggestion
      return pcloudSdk.createClient(
```

Reply:
Searched for all the `return await`s and fixed them - thanks for pointing this out, I should've paid more attention and searched when fixing the first time!